### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by otraore**
Fixes https://github.com/EngoEngine/engo/issues/284 Work In Progress

### CODE REVIEW

File: engo_test.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+func TestSetOverrideCloseAction(t *testing.T) {
+    SetOverrideCloseAction(true)
+    assert.Equal(t, opts.OverrideCloseAction, true)
+    SetOverrideCloseAction(false)
+    assert.Equal(t, opts.OverrideCloseAction, false)
+}
+
+func TestSetFPSLimit(t *testing.T) {
+    err := SetFPSLimit(60)
+    assert.Equal(t, err, nil)
```
Comment (EtienneBruines) [inline engo_test.go]: `assert.Nil(t, err)`, IIRC

File: engo_test.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    assert.NotEqual(t, err, nil)
+}
+
+func TestHeadless(t *testing.T) {
+    SetHeadless(false)
+    assert.Equal(t, opts.HeadlessMode, false)
+    assert.Equal(t, Headless(), false)
+    SetHeadless(true)
+    assert.Equal(t, opts.HeadlessMode, true)
+    assert.Equal(t, Headless)
```
Comment (EtienneBruines) [inline engo_test.go]: This is missing something.

File: engo_test.go [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    err := SetFPSLimit(60)
+    assert.Equal(t, err, nil)
+    assert.Equal(t, opts.FPSLimit, 60)
+    err = SetFPSLimit(-1)
+    assert.NotEqual(t, err, nil)
+}
+
+func TestHeadless(t *testing.T) {
+    SetHeadless(false)
+    assert.Equal(t, opts.HeadlessMode, false)
```
Comment (EtienneBruines) [inline engo_test.go]: `assert.False(t, HeadlessMode)` (as with everything else)

File: engo_test.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    assert.Equal(t, Headless(), false)
+    SetHeadless(true)
+    assert.Equal(t, opts.HeadlessMode, true)
```
Comment (EtienneBruines) [inline engo_test.go]: `assert.True(t, opts.HeadlessMode)` (as with everything else)

File: engo_glfw_test.go [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+func TestWindowSize(t *testing.T) {
+    assert.Equal(t, WindowSize(), window.GetSize())
+    assert.Equal(t, WindowWidth(), windowWidth)
+    assert.Equal(t, WindowHeight(), windowHeight)
+}
+
+func TestCanvasSize(t *testing.T) {
+    assert.Equal(t, CanvasWidth(), canvasWidth)
+    assert.Equal(t, CanvasHeight(), canvasHeight)
+}
```
Comment (EtienneBruines) [inline engo_glfw_test.go]: While these tests are nice, they are more meaningful when a window is actually created. (haven't checked why travis fails, but I guess this is why)

`TestCanvasSize` vs `TestGameWidthHeight` (I prefer the `-Size` version).

File: engo_glfw_test.go [hunk 6]
```diff
@@ -0,0 +1,29 @@
+//+build !netgo,!android
+
+package engo
+
+import (
+    "testing"
+    
+    "github.com/github.com/stretchr/testify/assert"
```
Comment (EtienneBruines) [inline engo_glfw_test.go]: :laughing:

**Comment by EtienneBruines** [comment]
To quote someone I know: 

> Documentation

File: engo_test.go [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    err = SetFPSLimit(-1)
+    assert.NotNil(t, err)
+}
+
+func TestHeadless(t *testing.T) {
+    SetHeadless(false)
+    assert.False(t, opts.HeadlessMode)
+    assert.False(t, Headless())
+    SetHeadless(true)
+    assert.True(t, opts.HeadlessMode, )
```
Comment (EtienneBruines) [inline engo_test.go]: `,` too much.

File: engo_test.go [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    SetScaleOnResize(false)
+    assert.False(t, opts.ScaleOnResize)
+    assert.False(t, ScaleOnResize())
+}
+
+func TestSetOverrideCloseAction(t *testing.T) {
+    SetOverrideCloseAction(true)
+    assert.True(t, opts.OverrideCloseAction)
+    SetOverrideCloseAction(false)
+    assert.True(t, opts.OverrideCloseAction)
```
Comment (EtienneBruines) [inline engo_test.go]: `assert.False`, I would presume.

File: engo_test.go [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    assert.NotNil(t, err)
+}
+
+func TestHeadless(t *testing.T) {
+    SetHeadless(false)
+    assert.False(t, opts.HeadlessMode)
+    assert.False(t, Headless())
+    SetHeadless(true)
+    assert.True(t, opts.HeadlessMode, )
+    assert.Equal(t, Headless)
```
Comment (EtienneBruines) [inline engo_test.go]: And this one still isn't `assert.True`.

File: engo_glfw_test.go [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+
+package engo
+
+import (
+	"testing"
+
+	"github.com/github.com/stretchr/testify/assert"
+)
+
+func TestGameSize(t *testing.T) {
```
Comment (paked) [inline engo_glfw_test.go]: These tests considering that the functions can only return this value

https://github.com/EngoEngine/engo/blob/master/engo_glfw.go#L267-L269

File: engo_glfw_test.go [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+	assert.Equal(t, GameWidth(), gameWidth)
+	assert.Equal(t, GameHeight(), gameHeight)
+}
+
```
Comment (paked) [inline engo_glfw_test.go]: Same here

File: engo_glfw_test.go [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+func TestCursorPos(t *testing.T) {
+	assert.Equal(t, CursorPos(), window.GetCursorPos())
+}
+
+func TestWindowSize(t *testing.T) {
```
Comment (paked) [inline engo_glfw_test.go]: And here

File: engo_glfw_test.go [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+	assert.Equal(t, WindowSize(), window.GetSize())
+	assert.Equal(t, WindowWidth(), windowWidth)
+	assert.Equal(t, WindowHeight(), windowHeight)
+}
+
+func TestCanvasSize(t *testing.T) {
```
Comment (paked) [inline engo_glfw_test.go]: annnd here

File: engo_glfw_test.go [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+	assert.Equal(t, CanvasWidth(), canvasWidth)
+	assert.Equal(t, CanvasHeight(), canvasHeight)
+}
```
Comment (paked) [inline engo_glfw_test.go]: I think the most valuable test we can do here is ensuring that the CanvasWidth and WindowWidth are correct given a certain scale.

File: engo_test.go [hunk 15]
```diff
@@ -0,0 +1,46 @@
+package engo
+
+import (
+	"testing"
+
+	"github.com/github.com/stretchr/testify/assert"
+)
+
+func TestSetScaleOnResize(t *testing.T) {
```
Comment (paked) [inline engo_test.go]: These tests literally test whether variable assignment works in Go

File: engo_test.go [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+func TestHeadless(t *testing.T) {
+	SetHeadless(false)
+	assert.False(t, opts.HeadlessMode)
+	assert.False(t, Headless())
+	SetHeadless(true)
+	assert.True(t, opts.HeadlessMode)
+	assert.Equal(t, Headless)
+}
+
+func TestExit(t *testing.T) {
```
Comment (paked) [inline engo_test.go]: This is a good test.

File: engo_glfw_test.go (hunk 10 above)
Comment (otraore) [inline engo_glfw_test.go]: That's the point of the tests to make sure the given function is doing what it's supposed to do. Same for all below.

File: engo_test.go (hunk 15 above)
Comment (otraore) [inline engo_test.go]: Testing that said function is doing what it's supposed to. If it did anything else, these will fill.

File: engo_glfw_test.go (hunk 14 above)
Comment (otraore) [inline engo_glfw_test.go]: I'll try to add this in the next round of commits

**Pull request closed without merging**